app/models/supplejack_api/top_metric.rb

```diff
@@ -66,5 +70,58 @@ def self.record_metrics_to_be_processed(date, metric)
         :processed_by_top_metrics.in => [nil, '', false]
       ).order_by(metric => 'desc').limit(200)
     end
+
+    def self.record_collection_metrics_to_be_processed(date, metric, collection)
+      SupplejackApi::RecordMetric.where(
+        date: date,
+        display_collection: collection,
+        :processed_by_top_collection_metrics.in => [nil, '', false]
+      ).order_by(metric => 'desc').limit(200)
+    end
+
+    #
+    # This method acts similarly to the ::spawn method defined above.
```

Comment:
Do you think you can move some lines that are the same on both methods? Make it a little `DRY`.

Reply:
yeah richard and i discussed this but it is quite complex. i think i do need to come back to this.

Reply:
refactored.